packages/terra-clinical-modal-manager/src/ModalManager.jsx

```diff
@@ -0,0 +1,198 @@
+   **/
+  children: PropTypes.node,
+
+  /**
+   * From `connect`. The Array of component keys that will be used to instantiate the Modal's inner components.
+   **/
+  modalComponentKeys: PropTypes.array,
+
+  /**
+   * From `connect`. An Object containing component data used to instantiate the Modal's inner components.
+   **/
+  modalComponentData: PropTypes.object,
+
+  /**
+   * From `connect`. The desired size of the modal.
+   **/
+  size: PropTypes.oneOf(['tiny', 'small', 'medium', 'large', 'huge']),
+
+  /**
+   * From `connect`. The presentation state of the modal.
+   **/
+  isOpen: PropTypes.bool,
+
+  /**
+   * From `connect`. The maximization state of the modal.
+   **/
+  isMaximized: PropTypes.bool,
+
+  /**
+   * From `connect`. A function that dispatches an `open` action.
+   **/
+  openModal: PropTypes.func.isRequired,
+
+  /**
+   * From `connect`. A function that dispatches a `close` action.
+   **/
+  closeModal: PropTypes.func.isRequired,
+
+  /**
+   * From `connect`. A function that dispatches a `push` action.
+   **/
+  pushModal: PropTypes.func.isRequired,
+
+  /**
+   * From `connect`. A function that dispatches a `pop` action.
+   **/
+  popModal: PropTypes.func.isRequired,
+
+  /**
+   * From `connect`. A function that dispatches a `maximize` action.
+   **/
+  maximizeModal: PropTypes.func.isRequired,
+
+  /**
+   * From `connect`. A function that dispatches a `minimize` action.
+   **/
+  minimizeModal: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  isOpen: false,
+  isMaximized: false,
+  size: 'small',
+};
+
+class ModalManager extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // I'm tracking the responsive-fullscreen state outside of React and Redux state to limit the number of
+    // renderings that occur.
+    this.forceFullscreenModal = false;
+
+    this.updateFullscreenState = this.updateFullscreenState.bind(this);
+    this.buildModalComponents = this.buildModalComponents.bind(this);
+  }
+
+  componentDidMount() {
+    this.updateFullscreenState();
+    window.addEventListener('resize', this.updateFullscreenState);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateFullscreenState);
+  }
+
+  updateFullscreenState() {
+    const previousFullscreenState = this.forceFullscreenModal;
+
+    this.forceFullscreenModal = window.innerWidth < getBreakpoints().small;
+
+    // Only update the modal if it's minimized, open, and changing states.
+    if (!this.props.isMaximized && this.props.isOpen && previousFullscreenState !== this.forceFullscreenModal) {
+      this.forceUpdate();
+    }
+  }
+
+  buildModalComponents() {
+    if (!this.props.modalComponentKeys || !this.props.modalComponentKeys.length) {
+      return null;
+    }
+
+    return this.props.modalComponentKeys.map((componentKey, index) => {
+      const componentData = this.props.modalComponentData[componentKey];
```

Comment:
What happens if modalComponentData does not contain the key? Codes this cause an error in line 126?